Oh, Gosh.  This isn't your best work.  I can see you haven't understood what I want here.  I've fixed makefile and VERBOSE.  However, every existing tests fails.  It has something to do with the '++ ' that you're adding to every command.  VALGRIND is hardly ever set.

```
======================================================================
ERROR: test_run (bugfixes.github.test_CVE_2017_1000126.TestCvePoC)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/rmills/gnu/github/exiv2/0.27-maintenance/tests/system_tests.py", line 588, in test_run
    proc = subprocess.Popen(
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/subprocess.py", line 854, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/subprocess.py", line 1702, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: '++'
```



Have you built and run the tests on 0.27-maintenance.  It behaves something like this:

```bash
$ cd build
$ make python_tests
.........s.......s.............................................s................................................ss...................
171 tests run in 3.02 seconds (5 skipped)
```
When VERBOSE is set:
```bash
$ cd build
$ make python_tests VERBOSE=1
addmoddel_test (bash_tests.testcases.TestCases) ... ok
conversions_test (bash_tests.testcases.TestCases) ... ok
....
171 tests run in 3.02 seconds (5 skipped)
```

When VERBOSE and EXIV2_ECHO are set:

```bash
$ cd build
$ make python_tests VERBOSE=1
addmoddel_test (bash_tests.testcases.TestCases) ...
/Users/rmills/gnu/github/exiv2/0.27-maintenance/build/bin/addmoddel exiv2-empty.jpg
ok
conversions_test (bash_tests.testcases.TestCases) ... 
/Users/rmills/gnu/github/exiv2/0.27-maintenance/build/bin/exiv2 -pv exiv2-empty.jpg
ok
....
171 tests run in 3.02 seconds (5 skipped)
```

--- a/test/Makefile
+++ b/test/Makefile
@@ -153,8 +153,8 @@ python_tests:
 	@echo
 	@echo ---- Running python_tests ----
 	@echo
-	@echo bash -c 'cd ../tests; export VERBOSE=1; python3 runner.py --verbose'
-	@bash -c 'cd ../tests; export VERBOSE=1; python3 runner.py --verbose'
+	@echo bash -c 'cd ../tests; python3 runner.py --verbose'
+	@bash      -c 'cd ../tests; python3 runner.py --verbose'
 
 mostlyclean clean:
 	rm -rf $(top_srcdir)/test/tmp/*

--- a/test/Makefile
+++ b/test/Makefile
@@ -153,8 +153,8 @@ python_tests:
 	@echo
 	@echo ---- Running python_tests ----
 	@echo
-	@echo bash -c 'cd ../tests; python3 runner.py --verbose'
-	@bash      -c 'cd ../tests; python3 runner.py --verbose'
+	@echo bash -c 'cd ../tests; python3 runner.py'
+	@bash      -c 'cd ../tests; python3 runner.py'
 
 mostlyclean clean:
 	rm -rf $(top_srcdir)/test/tmp/*

--- a/tests/bash_tests/utils.py
+++ b/tests/bash_tests/utils.py
@@ -461,7 +461,7 @@ class Executer:
 
     def run(self):
         # Whether to display the command to execute
-        if Config.exiv2_echo or Config.verbose:
+        if Config.exiv2_echo:
             print('++', ' '.join(self.args))
 
         # Check stdout

--- a/tests/bash_tests/utils.py
+++ b/tests/bash_tests/utils.py
@@ -32,7 +32,7 @@ class Config:
     exiv2_port          = '12760'
     exiv2_echo          = ''
     verbose             = ''
-    valgrind            = '++'
+    valgrind            = ''
 
     @classmethod
     def init(cls):
@@ -462,7 +462,7 @@ class Executer:
     def run(self):
         # Whether to display the command to execute
         if Config.exiv2_echo or Config.verbose:
-            print(Config.valgrind, ' '.join(self.args))
+            print('++', ' '.join(self.args))
 
         # Check stdout
         if self.redirect_stderr_to_stdout:

--- a/tests/runner.py
+++ b/tests/runner.py
@@ -62,9 +62,10 @@ if __name__ == '__main__':
         nargs='?'
     )
 
-    args = parser.parse_args()
-    conf_file = args.config_file[0]
-    DEFAULT_ROOT = os.path.abspath(os.path.dirname(conf_file))
+    args          = parser.parse_args()
+    args.verbose |=  "VERBOSE" in os.environ
+    conf_file     = args.config_file[0]
+    DEFAULT_ROOT  = os.path.abspath(os.path.dirname(conf_file))
     system_tests.set_debug_mode(args.debug)
     system_tests.configure_suite(conf_file)
 

--- a/tests/suite.conf
+++ b/tests/suite.conf
@@ -20,7 +20,7 @@ exiv2_http: http://127.0.0.1
 exiv2_port: 12760
 exiv2_echo: 
 verbose: 
-valgrind: ++
+valgrind:
 
 [paths]
 exiv2: ${ENV:exiv2_path}/exiv2

